geth/peers/peerpool.go

```diff
@@ -95,22 +96,55 @@ func (p *PeerPool) Start(server *p2p.Server) error {
 // handleServerPeers watches server peer events, notifies topic pools about changes
 // in the peer set and stops the discv5 if all topic pools collected enough peers.
 func (p *PeerPool) handleServerPeers(server *p2p.Server, events <-chan *p2p.PeerEvent) {
+	var (
+		toSearch    []*TopicPool
+		retryDiscv5 <-chan time.Time
+	)
+	runListener := func() {
+		if server.DiscV5 == nil {
+			ntab, err := StartDiscv5(server)
+			if err != nil {
+				log.Error("starting discv5 failed", "error", err)
+				retryDiscv5 = time.After(2 * time.Second)
+				return
+			}
+			log.Debug("restarted discovery from peer pool")
+			server.DiscV5 = ntab
+		}
+		for _, t := range toSearch {
+			_ = t.StartSearch(server)
```

Comment:
We should at least log errors.

Reply:
the only error is possible if discv5 isnt running, but sure i will add a log